src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1588,13 +1588,27 @@ public override BoundNode VisitCall(BoundCall node)
                 ImmutableArray<RefKind> refKindsOpt = node.ArgumentRefKindsOpt;
                 ImmutableArray<BoundExpression> arguments = RemoveArgumentConversions(node.Arguments, refKindsOpt);
                 ImmutableArray<int> argsToParamsOpt = node.ArgsToParamsOpt;
-                ImmutableArray<Result> results = VisitArgumentsEvaluate(arguments, refKindsOpt, method.Parameters, argsToParamsOpt, node.Expanded);
```

Comment:
>VisitArgumentsEvaluate [](start = 49, length = 22)

:memo: The problem here was that this was doing both `VisitArgumentsEvaluate` (the overload with the core logic) and `VisistArgumentsEvaluateHonoringAnnotations`. The latter could leave the state in split/conditional state. But `VisitArgumentsWarn` (invoke below) assumes that the state is not conditional (it calls `TrackNullableStateForAssignment` in the `RefKind.Out` case).

The new ordering moves `VisistArgumentsEvaluateHonoringAnnotations` after `VisitArgumentsWarn` (which was renamed).